pkg/vm/instruction.go

```diff
@@ -223,25 +224,28 @@ func decodeInstructionFlags(instruction *Instruction, flags uint16) error {
 	return nil
 }
 
-// Given []uint16 of 0s or 1s returns the set bit if there's only one such
-// and return len(bits) in case there's no set bits.
+// Given a uint16 returns the set bit offset by offset if there's only one such
+// and return maxLen in case there's no set bits.
 // If there are more than 1 set bits return an error.
-func oneHot(bits ...uint16) (uint16, error) {
-	var checkSum uint16 = 0
-	setBit := len(bits)
+func oneHot(bin uint16, offset, maxLen uint16) (uint16, error) {
+	numberOfBits := bits.OnesCount16(bin)
 
-	// checking
-	for i, bit := range bits {
-		checkSum += bit
+	if numberOfBits > 1 {
+		digits := make([]uint8, 0)
```

Comment:
You can directly specify the capacity of this slice, since you know it before execution:
```
make([]uint8, 0, maxLen)
```

Reply:
Though more of a nitpick, is not really that important